staging/src/k8s.io/legacy-cloud-providers/azure/auth/azure_auth.go

```diff
@@ -68,6 +68,16 @@ type AzureAuthConfig struct {
 	// ResourceManagerEndpoint is the cloud's resource manager endpoint. If set, cloud provider queries this endpoint
 	// in order to generate an autorest.Environment instance instead of using one of the pre-defined Environments.
 	ResourceManagerEndpoint string `json:"resourceManagerEndpoint,omitempty" yaml:"resourceManagerEndpoint,omitempty"`
+	// Configuration of creating and using network resources in different AAD Tenant than the one of the cluster
+	CrossTenantNetworkResourceConfig *CrossTenantNetworkResourceConfig `json:"crossTenantNetworkResource,omitempty" yaml:"crossTenantNetworkResource,omitempty"`
```

Comment:
This PR introduced a new struct here, what about using `string` as above, it would make config easier

Reply:
@feiskyer this PR only introduced two fields, use one level instead of one struct could be simpler and looks cleaner, WDYT?

Reply:
Yep, I think so

Reply:
Yep, I think so

Reply:
`CrossTenant` looks confusing. Could you rename them to `NetworkResourceTenantID` and `NetworkResourceSubscriptionID`?

Reply:
Sure. Thank you guys.